custom_parser/__init__.py

```diff
@@ -0,0 +1,27 @@
+"""Parser library.
+
+Each format of the input data should be parsed in different manner. This
+library provides custom parsers for each such usecase. The "parse"
+function used by data-collector is decided by env variable INPUT_DATA_FORMAT.
+"""
+
+import logging
+import os
+
+logger = logging.getLogger()
+
+if os.environ.get('INPUT_DATA_FORMAT') == 'CSV':
```

Comment:
That's correct observation. The data-collector stays locked to a specific data format in the same manner as it stays locked to the same AI service...

Like.. I can't think of any case, when we would like the particular deployment of the data-collector to handle different format than the one it already does.

We expect that the data we are receiving for the one particular AI service stays consistent... It may change in format, yes, but that would be a one-off change. The data collector is not intended to be handling many different formats at once. For one particular scheme, it does one job, however the pluggable parser makes it easier to redeploy the same base into a different pipeline.

-> We aim to deploy a separate pipeline for each usecase, otherwise we can't ensure the consumer can distinguish the outputs. So we can have data collectors locked to certain format.

Reply:
Ok, thanks for clarifying!